### ISSUE TITLE
Large library causes timeout (504)

### ISSUE THREAD

**Issue by zenjabba** [se1exin/Plex-Library-Cleaner#1]
```2020/02/09 17:17:17 [error] 12#12: *3 upstream timed out (110: Connection timed out) while reading response header from upstream, client: 172.17.0.1, server: , request: "GET /movies/dupes HTTP/1.1", upstream: "uwsgi://unix:///tmp/uwsgi.sock", host: "localhost:5000", referrer: "http://localhost:5000/"
172.17.0.1 - - [09/Feb/2020:17:17:17 +0000] "GET /movies/dupes HTTP/1.1" 504 569 "http://localhost:5000/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.87 Safari/537.36" "-"```

Looks like your doing standard timeout

**Comment by se1exin** [comment]
The timeout is 120 seconds - but obviously not enough for your library size.

Roughly how many items do you have? And do you have an idea of how many have duplicates (I know that's what this tool is supposed to tell you).

**Comment by se1exin** [comment]
I've changed the timeout to 1 hour - which should be more than enough! Please try it out and let me know how it goes.

**Comment by jorjenava** [comment]
I'm also having this issue. My library consists of 541 movies and 8841 individual tv episodes. Not sure how many of those are duplicates. Total library size is near 20tb.

**Comment by se1exin** [comment]
Thanks for the additional info @jorjenava. If you are able to, can you report if your plex is on a remote computer (e.g. on the same network vs over the internet), and what the operating system the server is on?

**Comment by jorjenava** [comment]
Server is on the same network. Server os is freenas 11.2. Plex is running in a jail. Client os is kubuntu 19.10 running latest Firefox. Client is on WiFi and server is hardwired.

**Comment by Suika** [comment]
PLC is timing out on a library that has 27766 entries.
Running official Plex Docker container, Version 1.21.1.3876
PLC latest version for dockerhub.

```
172.23.0.50 - - [18/Jan/2021:23:04:22 +0000] "GET / HTTP/1.1" 304 0 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.183 Safari/537.36" "10.0.0.102"
Respawned uWSGI worker 1 (new pid: 27)
[pid: 27|app: 0|req: 3/6] 172.23.0.50 () {60 vars in 988 bytes} [Mon Jan 18 23:04:26 2021] GET /favicon.ico => generated 15406 bytes in 4 msecs via sendfile() (HTTP/1.1 200) 7 headers in 277 bytes (0 switches on core 0)
172.23.0.50 - - [18/Jan/2021:23:04:26 +0000] "GET /favicon.ico HTTP/1.1" 200 15406 "https://plc.host.lan/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.183 Safari/537.36" "10.0.0.102"
Mon Jan 18 23:04:40 2021 - SIGPIPE: writing to a closed pipe/socket/fd (probably the client disconnected) on request /movies/dupes (ip 172.23.0.50) !!!
Mon Jan 18 23:04:40 2021 - uwsgi_response_writev_headers_and_body_do(): Broken pipe [core/writer.c line 306] during GET /movies/dupes (172.23.0.50)
OSError: write error
[pid: 15|app: 0|req: 3/7] 172.23.0.50 () {56 vars in 917 bytes} [Mon Jan 18 22:57:57 2021] GET /movies/dupes => generated 0 bytes in 403108 msecs (HTTP/1.1 200) 3 headers in 0 bytes (0 switches on core 0)
worker 2 killed successfully (pid: 15)
uWSGI worker 2 cheaped.
2021/01/18 23:05:22 [error] 11#11: *9 upstream timed out (110: Connection timed out) while reading response header from upstream, client: 172.23.0.50, server: , request: "GET /movies/dupes HTTP/1.1", upstream: "uwsgi://unix:///tmp/uwsgi.sock", host: "plc.host.lan", referrer: "https://plc.host.lan/"
172.23.0.50 - - [18/Jan/2021:23:05:22 +0000] "GET /movies/dupes HTTP/1.1" 504 570 "https://plc.host.lan/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.183 Safari/537.36" "10.0.0.102"
```

**Comment by se1exin** [comment]
FYI I've merged @Suika 's PR #15 which limits the Plex query to 50 dupes at a time. Thanks @Suika !

Please reopen this issue if timeouts persist

**Comment by vtechnoviking** [comment]
Please reopen, large library 5301 films in five separate libraries.
